main.nf

```diff
@@ -9,279 +9,812 @@
         <dl class=\"dl-horizontal\">
-${summary.collect { k,v -> "            <dt>$k</dt><dd><samp>${v ?: '<span style=\"color:#999999;\">N/A</a>'}</samp></dd>" }.join("\n")}
+            $x
         </dl>
-    """.stripIndent()
-
-   return yaml_file
-}
-
+    """.stripIndent() }
+    .set { ch_workflow_summary }
 
 /*
  * Parse software version numbers
  */
 process get_software_versions {
+    publishDir "${params.outdir}/pipeline_info", mode: 'copy',
+        saveAs: { filename ->
+                      if (filename.indexOf(".csv") > 0) filename
+                      else null
+                }
 
     output:
-    file 'software_versions_mqc.yaml' into software_versions_yaml
+    file 'software_versions_mqc.yaml' into ch_software_versions_yaml
+    file "software_versions.csv"
 
     script:
-    // TODO nf-core: Get all tools to print their version number here
     """
     echo $workflow.manifest.version > v_pipeline.txt
     echo $workflow.nextflow.version > v_nextflow.txt
-    fastqc --version > v_fastqc.txt
+    trim_galore --version > v_trimgalore.txt
+    slamdunk --version > v_slamdunk.txt
     multiqc --version > v_multiqc.txt
-    scrape_software_versions.py > software_versions_mqc.yaml
+    scrape_software_versions.py &> software_versions_mqc.yaml
     """
 }
 
+/*
+ * Check design
+ */
+process checkDesign {
+
+    input:
+    file (design) from checkChannel
+
+    output:
+    file "nfcore_slamseq_design.txt" into rawFileChannel,
+                                          deseq2ConditionChannel,
+                                          splitChannel,
+                                          vcfSampleChannel
+
+    script:
+    """
+    check_design.py ${design} nfcore_slamseq_design.txt
+    """
+}
 
+rawFileChannel
+   .splitCsv( header: true, sep: '\t' )
+   .map { row -> tuple(row, file(row.reads, checkIfExists: true) ) }
+   .set { rawFiles }
+
+splitChannel
+   .splitCsv( header: true, sep: '\t' )
+   .set { conditionDeconvolution }
+
+vcfSampleChannel
+   .splitCsv( header: true, sep: '\t' )
+   .map{it ->
+       return it.name
+   }
+   .combine(vcfChannel)
+   .set{ vcfCombineChannel }
+
+/*
+ * STEP 1 - TrimGalore!
+ */
+if (params.skip_trimming) {
+    rawFiles
+       .map{it ->
+        return tuple(it, file(it.reads))
+       }
+       .set{ trimmedFiles }
+    trimgaloreQC = Channel.empty()
+    trimgaloreFastQC = Channel.empty()
+} else {
+  process trim {
+
+       tag { parameters.name }
+
+       input:
+       set val(parameters), file(reads) from rawFiles
+
+       output:
+       set val(parameters), file("TrimGalore/${parameters.name}.fastq.gz") into trimmedFiles
+       file ("TrimGalore/*.txt") into trimgaloreQC
+       file ("TrimGalore/*.{zip,html}") into trimgaloreFastQC
+
+       script:
+       """
+       mkdir -p TrimGalore
+       trim_galore ${reads} --stringency 3 --fastqc --cores ${task.cpus} --output_dir TrimGalore
+       mv TrimGalore/*.fq.gz TrimGalore/${parameters.name}.fastq.gz
```

Comment:
I think you could just use the `--basename` argument for TrimGalore here, though you will still have the `_trimmed` suffix added (not sure that's a problem?)

Reply:
Nope that worked fine - thanks for the suggestion